src/hotspot/share/opto/superwordVTransformBuilder.cpp

```diff
@@ -139,9 +139,13 @@ VTransformVectorNode* SuperWordVTransformBuilder::make_vector_vtnode_for_pack(co
   VTransformVectorNode* vtn = nullptr;
 
   if (p0->is_Load()) {
-    vtn = new (_vtransform.arena()) VTransformLoadVectorNode(_vtransform, pack_size);
+    const VPointer& scalar_p = _vloop_analyzer.vpointers().vpointer(p0->as_Load());
+    const VPointer vector_p(scalar_p.make_with_size(scalar_p.size() * pack_size));
+    vtn = new (_vtransform.arena()) VTransformLoadVectorNode(_vtransform, pack_size, vector_p);
   } else if (p0->is_Store()) {
-    vtn = new (_vtransform.arena()) VTransformStoreVectorNode(_vtransform, pack_size);
+    const VPointer& scalar_p = _vloop_analyzer.vpointers().vpointer(p0->as_Store());
+    const VPointer vector_p(scalar_p.make_with_size(scalar_p.size() * pack_size));
+    vtn = new (_vtransform.arena()) VTransformStoreVectorNode(_vtransform, pack_size, vector_p);
```

Comment:
Note: we now pass in the `VPointer` of the size of the vector directly. Before, we referenced the `VPointer`s of the packed nodes. But that reliance will be in the way in the future, and it is more clean to carry the modified `VPointer` like this.